docs/api_pma/components.rst

```diff
@@ -0,0 +1,116 @@
+
+Button
+-------------------------------
+
+.. image:: ../_static/pma/foundation_kit/components/button.jpg
+
+A Button is a digital input device; it generates a boolean value whenever it's pressed or released.
```

Comment:
```suggestion
A Button is a binary digital input device; this means that it has 2 states - pressed or released. This class provides an easy way to query the state of the device as well as perform event-based actions when it changes state.
```

Reply:
What about if we create a page that explains the ports for each type of device, and link to that from "digital input device"? We can remove the need to explain the available ports for each component; or perhaps we can `.. include` it from somewhere so that it isn't copy/pasted everywhere?